### DISCUSSION TITLE
Create an alert with exported values

### DISCUSSION THREAD

**Discussion by Brightlystake** [grafana/grafana#75743] [General]
i had an alert created and exported it as yaml file. 
i also have a service account created which is retrieving values so thats allright.

with yaml file i tried to convert to json via python and tried to create it again
This is throwing 400 error
{"message":"invalid alert rule: no queries or expressions are found","traceID":""}

i looked at https://grafana.com/docs/grafana/latest/developers/http_api/alerting_provisioning/#provisioned-alert-rule

there are few mandatory items which was missing 
folderUID
OrgID
ruleGroup

wondering why this was not present in the exported yaml.

Also, is my approach correct or should i approach it differently. Please help me with this.

`import requests
import yaml
import urllib3
import json

# Your YAML configuration
yaml_config = """
apiVersion: 1
groups:
  - orgId: 1
    name: Upgrade_Alert
    folder: Upgrade
    interval: 1m
    rules:
      - uid: c9f00616-5088-4779-a4f0-ca2145288cbe
        title: upgrade
        condition: C
        data:
          - refId: A
            relativeTimeRange:
              from: 65
              to: 0
            datasourceUid: a98b90ea-6dfd-4ba7-b376-01e517d90767
            model:
              editorMode: code
              expr: cometbft_consensus_height{instance="XXXXXX",job="XXXXXX"}
              hide: false
              instant: true
              intervalMs: 1000
              maxDataPoints: 43200
              range: false
              refId: A
          - refId: B
            relativeTimeRange:
              from: 600
              to: 0
            datasourceUid: __expr__
            model:
              conditions:
                - evaluator:
                    params:
                      - 0
                      - 0
                    type: gt
                  operator:
                    type: and
                  query:
                    params: []
                  reducer:
                    params: []
                    type: avg
                  type: query
              datasource:
                name: Expression
                type: __expr__
                uid: __expr__
              expression: A
              intervalMs: 1000
              maxDataPoints: 43200
              reducer: last
              refId: B
              settings:
                mode: replaceNN
                replaceWithValue: 0
              type: reduce
          - refId: C
            datasourceUid: __expr__
            model:
              conditions:
                - evaluator:
                    params:
                      - 0
                      - 0
                    type: gt
                  operator:
                    type: and
                  query:
                    params: []
                  reducer:
                    params: []
                    type: avg
                  type: query
              datasource:
                name: Expression
                type: __expr__
                uid: __expr__
              expression: ' 1278000 - $B < 500'
              hide: false
              intervalMs: 1000
              maxDataPoints: 43200
              refId: C
              type: math
        noDataState: Alerting
        execErrState: Alerting
        for: 5m
        isPaused: false
"""

# Parse YAML
config_data = yaml.safe_load(yaml_config)

print('------')

# API endpoint
base_url = "https://monitoring.XXXXXXX.com/api/v1/provisioning/alert-rules"

# Authorization header
headers = {
    "Authorization": "Bearer glsa_XXXXXXXXXX",
    "Accept": "application/json",
    "Content-Type": "application/json",
}

# Make the request
response = requests.post(base_url, json=yaml_config, headers=headers, verify=False)

# Print the response
print(response.status_code)
print(response.text)
`

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move https://github.com/grafana/grafana/discussions/73424.

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!
